ionoscloud/resource_gpu_server.go

```diff
@@ -0,0 +1,1319 @@
+package ionoscloud
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	ionoscloud "github.com/ionos-cloud/sdk-go/v6"
+
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/bundleclient"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/cloudapi/cloudapifirewall"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/cloudapi/cloudapinic"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/cloudapi/cloudapiserver"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/cloudapi/nsg"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/slice"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils/constant"
+)
+
+func resourceGPUServer() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceGpuServerCreate,
+		ReadContext:   resourceGpuServerRead,
+		UpdateContext: resourceGpuServerUpdate,
+		DeleteContext: resourceGpuServerDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceGpuServerImport,
+		},
+		CustomizeDiff: checkServerImmutableFields,
+
+		Schema: map[string]*schema.Schema{
+			"template_uuid": {
```

Comment:
@avirtopeanu-ionos same question for the resource: if schema is the same or almost the same, can't we reuse some code?

Reply:
it is almost the same sadly. The NIC is optional for GPUs, the disk_type is computed etc